### PR TITLE
Preparatory clean-up of ARC nSIM board before adding ARC HS cores support

### PR DESCRIPTION
**Pull request by abrodkin**
What we do in this series:
1. Clean-up of copy-pastes from EMSK board
2. Rename of `nsim_em` board to more generic `nsim` so it becomes usable for ARC HS cores
3. Move CPU selection to defconfigs removing implicit selection of default something

### CODE REVIEW

**Review by mbolivar — APPROVED** [review]
LGTM

**Review by tyrantlizard77 — APPROVED** [review]

File: boards/arc/nsim/Kconfig.board [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+#
+# Copyright (c) 2016, 2019 Synopsys, Inc. All rights reserved.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+config BOARD_NSIM
+	bool "ARC nSIM simulator"
+	depends on SOC_NSIM
+	help
+	  The DesignWare ARC nSIM board is a virtual board based on
```
Comment (dbkinder) [inline boards/arc/nsim/Kconfig.board]: How about:
```
          The DesignWare ARC nSIM board is a virtual board based on
	  the ARC nSIM simulator.  It demonstrates the ARC core features
	  and a console based on the hostlink feature of the nSIM simulator.
```

**Review by dbkinder — CHANGES_REQUESTED** [review]

File: boards/arc/nsim/Kconfig.board (hunk 1 above)
Comment (abrodkin) [reply boards/arc/nsim/Kconfig.board]: Well nSIM simulates among other things legacy ARC UART which is used for serial console in Zephyr/Linux/U-Boot and this is a completely different beast as compared to so-called hostlink (which is just a mailbox in target's memory).

Makes sense?

Comment (dbkinder) [reply boards/arc/nsim/Kconfig.board]: I was trying to clarify the English a bit.  You wrote, "It just has a console based on hostlink feature of nSIM simulator."  Is what I suggested incorrect?  Can you use the simulated UART for console output?

**Review by dbkinder — CHANGES_REQUESTED** [review]

File: boards/arc/nsim/Kconfig.board (hunk 1 above)
Comment (abrodkin) [reply boards/arc/nsim/Kconfig.board]: @dbkinder Oops, sorry. I blindly fixed spelling of `nSIM` in original description (https://github.com/zephyrproject-rtos/zephyr/blob/master/boards/arc/nsim_em/Kconfig.board) and failed to realize description by its nature was misleading. Indeed we use simulated UART (ARC UART, see https://github.com/zephyrproject-rtos/zephyr/blob/master/drivers/serial/Kconfig.nsim).

So based on your suggestion I'll reword the comment like that:
> The DesignWare ARC nSIM board is a virtual board based on
> the ARC nSIM simulator.  It demonstrates the ARC core features
> and a console based on the legacy ARC UART model.

**Comment by abrodkin** [comment]
@dbkinder pushed updated version. And thanks for the feedback!

**Review by vonhust — APPROVED** [review]
I am ok. It's a necessary cleanup before we add ARC HS support

**Review by ruuddw — APPROVED** [review]
I expect a lot of commonality between the nSIM EM and HS targets, makes sense to generalize instead of duplicate.

**Comment by abrodkin** [comment]
Yet another respin with rework of device tree description for:
1. Board-specific compatible & model
2. Ability to reuse .dtsi for other nSIM board flavors (read to be added ARC HS variants)

**Review by dbkinder — APPROVED** [review]
LGTM, thanks!

**Comment by nashif** [comment]
issues in ci, for example:

```
In file included from ../../../../../../include/kernel_includes.h:21,
                 from ../../../../../../include/kernel.h:17,
                 from ../../../../../../kernel/include/kernel_structs.h:10,
                 from /home/buildslave/src/github.com/zephyrproject-rtos/zephyr/arch/arc/core/offsets/offsets.c:26:
../../../../../../include/atomic.h:94:28: note: previous definition of 'atomic_sub' was here
 static inline atomic_val_t atomic_sub(atomic_t *target, atomic_val_t value)
```

**Comment by abrodkin** [comment]
@nashif  

> issues in ci, for example:
> 
> ```
> In file included from ../../../../../../include/kernel_includes.h:21,
>                  from ../../../../../../include/kernel.h:17,
>                  from ../../../../../../kernel/include/kernel_structs.h:10,
>                  from /home/buildslave/src/github.com/zephyrproject-rtos/zephyr/arch/arc/core/offsets/offsets.c:26:
> ../../../../../../include/atomic.h:94:28: note: previous definition of 'atomic_sub' was here
>  static inline atomic_val_t atomic_sub(atomic_t *target, atomic_val_t value)
> ```

Silly me. That was due to the following hunk that was supposed to get in later (and in a different form when adding ARC HS support):
```diff
--- a/arch/arc/Kconfig
+++ b/arch/arc/Kconfig

+config CPU_ARCHS
+       bool
+       default y
+       select CPU_ARCV2
+       select ATOMIC_OPERATIONS_BUILTIN
+       help
+         This option signifies the use of an ARC HS CPU
 endmenu
```

Given it was a copy-paste of `CPU_ARCEM` it has 2 things which when combined together give us the problem:
1. "default y" selecting this option as it was situated below `CPU_ARCEM`
2. select `ATOMIC_OPERATIONS_BUILTIN` which selects GCC's built-in atomic ops which are missing when building with `-mcpu=em4_fpus` as ARC EM family doesn't support `llock`/`scond` atomics.

Will push a fixed respin.

**Pull request merged**